pkg/controller/daemon/daemon_controller.go

```diff
@@ -801,6 +801,68 @@ func (dsc *DaemonSetsController) resolveControllerRef(namespace string, controll
 	return ds
 }
 
+// syncPodOnNode sync the DaemonSet Pods on the node, the return values are:
+//   - nodesNeedingDaemonPods: the pods need to start on the node
+//   - podsToDelete: the Pods need to be deleted on the node
+func (dsc *DaemonSetsController) syncPodOnNode(
+	node *v1.Node,
+	nodeToDaemonPods map[string][]*v1.Pod,
+	ds *extensions.DaemonSet,
+) (nodesNeedingDaemonPods, podsToDelete []string, failedPodsObserved int, err error) {
+
+	wantToRun, shouldSchedule, shouldContinueRunning, err := dsc.nodeShouldRunDaemonPod(node, ds)
```

Comment:
I assume after all your changes are done, we will only get `wantToRun` and we will eliminate `shouldSchedule`. Basically, DS controller won't check whether node can run the Pod and leaves that decision to the scheduler.

Reply:
yes. The target is to remove it, I think we will start this feature from alpha, so maybe we will keep it as is for 1.10.

Reply:
SGTM